packages/contracts/src/utils/protocol/VersionComparisonLib.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.8;
+
+library VersionComparisonLib {
```

Comment:
Is there a strong reason for: why we use a library?
- we have to pay for library deploy, then add it to DAO's contracts.
- it will increase the size of the DAO's contract by including all the functionlities that are not used as of now

why not we are using free functions?

Reply:
I tested this:

A library with two internal functions (addOne, subOne) and one is used: 116,715 gas
https://goerli.etherscan.io/address/0x73C95A0e6c655974adf3bCFF82b0BAe378c95fE8#code

A library with two internal functions (addOne, subOne) and both are used: 138,471 gas
https://goerli.etherscan.io/address/0xca52c84593c77ff0381648b59d9fea82d921d848#code

A library with one internal function (addOne) and one is used: 116,703 gas
https://goerli.etherscan.io/address/0x7d68eDb8A0D62D7149d58e47713846f8a332DBA4#code

Accordingly, the compiler is smart and only includes internal functions that are used in the contract bytecode.
We should use more libraries (see [uniswap-v3](https://github.com/Uniswap/v3-core/tree/main/contracts/libraries)).